src/config.rs

```diff
@@ -8,27 +8,31 @@ use std::fs;
 use std::path::PathBuf;
 
 pub const DB_MAX_SIZE: u64 = 8 * 1024 * 1024; // 8 MiB
-
 pub const DB_VERSION: DBVersion = 3;
 
-pub fn zo_data() -> Result<PathBuf> {
-    let path = match env::var_os("_ZO_DATA") {
-        Some(data_osstr) => PathBuf::from(data_osstr),
+pub fn zo_data_dir() -> Result<PathBuf> {
+    match env::var_os("_ZO_DATA_DIR") {
+        Some(data_osstr) => {
+            let data_dir = PathBuf::from(data_osstr);
+
+            if data_dir.is_file() {
```

Comment:
There's a problem here - even if `data_dir` is not a file, it still need not be a directory (eg. a symlink to a file).

Instead, I'd suggest returning a `PathBuf` from both branches of the `match` expression, and performing `fs::create_dir_all` on it (which should error out if `data_dir` is anything but a directory).

Reply:
I'm not entirely grokking the problem (through not fault of your own -- a little tired at the moment). I'll revisit this in the morning.

Reply:
OK, I understand now. I added a comment above the `fs::create_dir_all` call to document its use.